stablehlo/reference/Element.cpp

```diff
@@ -173,6 +173,221 @@ bool areApproximatelyEqual(APFloat f, APFloat g) {
   return std::fabs(f.convertToDouble() - g.convertToDouble()) <= 0.0001;
 }
 
+APInt leftShiftAndMask(APInt binary, int64_t shift, APInt mask) {
+  binary = binary.shl(shift);
+  binary |= mask;
+  return binary;
+}
+
+// Returns bitcast converted array of smaller bitwidth Element objects to one
+// larger bitwidth Element object.
+Element bitcastConvertManyToOne(Type type, ArrayRef<Element> elements) {
+  SmallVector<Element> results;
+  auto operandType = elements[0].getType();
+  int64_t operandBitWidth = isSupportedComplexType(operandType)
+                                ? operandType.cast<ComplexType>()
+                                      .getElementType()
+                                      .cast<FloatType>()
+                                      .getIntOrFloatBitWidth()
+                                : operandType.getIntOrFloatBitWidth();
+  int64_t resultBitWidth = isSupportedComplexType(type)
+                               ? type.cast<ComplexType>()
+                                     .getElementType()
+                                     .cast<FloatType>()
+                                     .getIntOrFloatBitWidth()
+                               : type.getIntOrFloatBitWidth();
+  if (isSupportedBooleanType(operandType) ||
+      isSupportedIntegerType(operandType) ||
+      isSupportedFloatType(operandType)) {
+    auto binary = APInt(resultBitWidth, 0);
+    for (auto element : llvm::reverse(elements)) {
+      APInt mask;
+      if (isSupportedBooleanType(operandType))
+        mask = APInt(resultBitWidth, element.getBooleanValue());
+      else if (isSupportedIntegerType(operandType))
+        mask = element.getIntegerValue().zext(resultBitWidth);
+      else if (isSupportedFloatType(operandType))
+        mask = element.getFloatValue().bitcastToAPInt().zext(resultBitWidth);
+      binary = leftShiftAndMask(binary, operandBitWidth, mask);
+    }
+
+    if (isSupportedBooleanType(type)) return elements[0];
```

Comment:
Given that numbits(elements[0].getType()) < type, IMO, type cannot be boolean. Seems like dead check to me. Can  you please double check?